File: C++/1_Creational Patterns/1_Singleton/Logger.cpp
#include <iostream>

class Logger {
    private:
        static Logger* instance;
        Logger() {}  // private constructor to prevent instantiation

    public:
        static Logger* getInstance() {
            if (instance == NULL) {
                instance = new Logger();
            }
            return instance;
        }

        void log(std::string message) {
            // implementation of logging logic goes here
            std::cout << "Logging: " << message << std::endl;
        }
};

Logger* Logger::instance = 0;

int main() {
    Logger* logger = Logger::getInstance();

    logger->log("Starting MyApp");

    // application logic goes here

    logger->log("Shutting down MyApp");

    return 0;
}

File: Java/1_Creational Patterns/1_Singleton/Logger.java
public class Logger {
    private static Logger instance = null;

    private Logger() {
        // private constructor to prevent instantiation
    }

    public static Logger getInstance() {
        if (instance == null) {
            instance = new Logger();
        }
        return instance;
    }

    public void log(String message) {
        // implementation of logging logic goes here
        System.out.println("Logging: " + message);
    }
}

public class MyApp {
    private static final Logger logger = Logger.getInstance();

    public static void main(String[] args) {
        logger.log("Starting MyApp");

        // application logic goes here

        logger.log("Shutting down MyApp");
    }
}